### PR TITLE
Allow multiple category ids, lat / long search for Algolia services endpoint

### PR DESCRIPTION
**Pull request by drcaramelsyrup**

### CODE REVIEW

File: postman/AskDarcel%20API.postman_collection.json [hunk 1]
```diff
@@ -517,7 +517,7 @@
 				"method": "GET",
 				"header": [],
 				"url": {
-					"raw": "{{base_url}}/services/search?site_id=1",
+					"raw": "{{base_url}}/services?site_id=1",
```
Comment (wi11) [inline postman/AskDarcel%20API.postman_collection.json:520]: Why are we removing the /search suffix here from the URL? This test is intended to test the algolia endpoint which uses the /search suffix right?

File: postman/AskDarcel%20API.postman_collection.json [hunk 2]
```diff
@@ -569,7 +569,7 @@
 				"method": "GET",
 				"header": [],
 				"url": {
-					"raw": "{{base_url}}/services/search?eligibility_id=1&site_id=1",
+					"raw": "{{base_url}}/services?eligibility_id=1&site_id=1",
```
Comment (wi11) [inline postman/AskDarcel%20API.postman_collection.json:572]: same as above

File: postman/AskDarcel%20API.postman_collection.json (hunk 1 above)
Comment (wi11) [inline postman/AskDarcel%20API.postman_collection.json:520]: Why are we removing the /search suffix here from the URL? This test is intended to test the algolia endpoint which uses the /search suffix right?

File: postman/AskDarcel%20API.postman_collection.json [hunk 3]
```diff
@@ -593,6 +593,174 @@
 			},
 			"response": []
 		},
+		{
+			"name": "Get services by category using algolia endpoint",
```
Comment (wi11) [inline postman/AskDarcel%20API.postman_collection.json:597]: nit: "Get services by multiple categories using algolia endpoint"

**Review by wi11 — CHANGES_REQUESTED** [review]

File: app/controllers/services_controller.rb [hunk 4]
```diff
@@ -141,23 +141,57 @@ def site_code
     params[:site_id] ? Site.find_by(id: params[:site_id]).site_code : "sfsg"
   end
 
-  def eligibility_name
-    params[:eligibility_id] ? Eligibility.find_by(id: params[:eligibility_id].to_i).name : ""
+  def eligibility_names
+    eligibilities = params[:eligibility_id].split(",")
+    Eligibility.where(["id in (?)", eligibilities]).map(&:name)
```
Comment (wi11) [inline app/controllers/services_controller.rb:146]: Can this line be simplified to this?

```suggestion
    Eligibility.where(id: eligibilities).map(&:name)
```

File: app/controllers/services_controller.rb [hunk 5]
```diff
@@ -141,23 +141,57 @@ def site_code
...
+  end
+
+  def category_names
+    categories = params[:category_id].split(",")
+    Category.where(["id in (?)", categories]).map(&:name)
```
Comment (wi11) [inline app/controllers/services_controller.rb:151]: Same as above

```suggestion
    Category.where(id: categories).map(&:name)
```

File: postman/AskDarcel%20API.postman_collection.json (hunk 1 above)
Comment (drcaramelsyrup) [reply postman/AskDarcel%20API.postman_collection.json]: Good catch, must have forgotten to save in the postman client.

File: postman/AskDarcel%20API.postman_collection.json (hunk 3 above)
Comment (drcaramelsyrup) [reply postman/AskDarcel%20API.postman_collection.json]: updated this.

File: app/controllers/services_controller.rb (hunk 4 above)
Comment (drcaramelsyrup) [reply app/controllers/services_controller.rb:146]: Looks like that does work with multiples too in my postman client! I'll update.

**Review by wi11 — APPROVED** [review]

**Comment by drcaramelsyrup** [comment]
I found that the multiple categories test had a bug in it while trying to figure out why the geolocation ranking wasn't passing the test consistently. It looks like `Model.where` doesn't seem to guarantee the services are returned in the order of the input ids. This latest commit addresses that with a `filter_map` (and ensuring existence / loading the `Service` data first).

As for the geoloc ranking, from Algolia's perspective, it looks like it _is_ working properly, just that the lat/lng in Algolia indices don't seem to match the generated lat/lng data stored in the service / resource addresses (from the `_rankingInfo` results):
```
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>676, "geoDistance"=>503, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7824, "lng"=>-122.443, "distance"=>503}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1263, "geoDistance"=>1121, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7747, "lng"=>-122.444, "distance"=>1121}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1261, "geoDistance"=>2462, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7837, "lng"=>-122.42, "distance"=>2462}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1522, "geoDistance"=>3151, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7769, "lng"=>-122.414, "distance"=>3151}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>595, "geoDistance"=>3387, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7649, "lng"=>-122.419, "distance"=>3387}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1539, "geoDistance"=>3575, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7785, "lng"=>-122.408, "distance"=>3575}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>646, "geoDistance"=>3655, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7955, "lng"=>-122.409, "distance"=>3655}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1129, "geoDistance"=>3729, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7817, "lng"=>-122.406, "distance"=>3729}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>2246, "geoDistance"=>4087, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7628, "lng"=>-122.411, "distance"=>4087}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>2255, "geoDistance"=>4432, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7881, "lng"=>-122.398, "distance"=>4432}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>614, "geoDistance"=>4739, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7477, "lng"=>-122.421, "distance"=>4739}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1172, "geoDistance"=>4835, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7871, "lng"=>-122.393, "distance"=>4835}}
api_1      | {"nbTypos"=>0, "firstMatchedWord"=>0, "proximityDistance"=>0, "userScore"=>1168, "geoDistance"=>4835, "geoPrecision"=>1, "nbExactWords"=>0, "words"=>0, "filters"=>3, "matchedGeoLocation"=>{"lat"=>37.7871, "lng"=>-122.393, "distance"=>4835}}
```

**Pull request merged**